modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOST.scala

```diff
@@ -0,0 +1,151 @@
+
+import cats.data.Reader
+import cats.data.EitherT
+import cats.effect.Sync
+import cats.implicits._
+import edu.gemini.spModel.config2.Config
+import edu.gemini.spModel.seqcomp.SeqConfigNames._
+import edu.gemini.spModel.gemini.ghost.Ghost
+import gem.math.{Angle, HourAngle}
+import scala.concurrent.duration._
+import seqexec.model.dhs.ImageFileId
+import seqexec.model.Model.{Instrument, Resource}
+import seqexec.server.ConfigUtilOps._
+import seqexec.server._
+import seqexec.server.keywords.{GDSClient, GDSInstrument}
+import seqexec.server.ghost.GHOSTController._
+import squants.time.{Seconds, Time}
+
+final case class GHOST[F[_]: Sync](controller: GHOSTController[F])
+    extends InstrumentSystem[F]
+    with GDSInstrument {
+  override val gdsClient: GDSClient = controller.gdsClient
+
+  override val resource: Resource = Instrument.GHOST
+
+  override val sfName: String = "GHOST"
+
+  override val contributorName: String = "ghost"
+
+  override val observeControl: InstrumentSystem.ObserveControl =
+    InstrumentSystem.Uncontrollable
+
+  override def observe(
+      config: Config): SeqObserveF[F, ImageFileId, ObserveCommand.Result] =
+    Reader { fileId =>
+      controller
+        .observe(fileId)
+        .map(_ => ObserveCommand.Success: ObserveCommand.Result)
+    }
+
+  override def configure(config: Config): SeqActionF[F, ConfigResult[F]] =
+    GHOST
+      .fromSequenceConfig[F](config)
+      .flatMap(controller.applyConfig)
+      .map(_ => ConfigResult[F](this))
+
+  override def notifyObserveEnd: SeqActionF[F, Unit] = controller.endObserve
+
+  override def notifyObserveStart: SeqActionF[F, Unit] = SeqActionF.void
+
+  override def calcObserveTime(config: Config): Time = Seconds(360)
+}
+
+object GHOST {
+  val INSTRUMENT_NAME_PROP: String = "GHOST"
+  val name: String                 = INSTRUMENT_NAME_PROP
+
+  val sfName: String = "GHOST"
+
+  // TODO: I am sure there is a cleaner way to write this without using for, but I'm not sure right
+  // TODO: now what it is.
+//  private def hmsAngle(config: Config, name: String): Either[ExtractFailure, Option[HourAngle]] =
+//    for {
+//      angOptStr <- config.extract(INSTRUMENT_KEY / name).as[Option[String]]
+//      angStr    <- angOptStr
+//      ang       <- HourAngle.fromStringHMS.getOption(angStr)
+//    } yield ang
+//  private def hmsAngle(config: Config, name: String): Either[ExtractFailure, Option[HourAngle]] =
+//    for {
+//      a <- config.extract(INSTRUMENT_KEY / name).as[String]
+//          .map(HourAngle.fromStringHMS.getOption)
+//    } yield a
+//
+//  private def dmsAngle(config: Config, name: String): Either[ExtractFailure, Option[Angle]] =
+//    for {
+//      a <- config.extract(INSTRUMENT_KEY / name).as[String]
+//        .map(Angle.fromStringSignedDMS.getOption)
+//    } yield a
+//
+//
+//  private def targetName(config: Config, name: String): Either[ExtractFailure, Option[String]] =
+//    config.extract(INSTRUMENT_KEY / name).as[Option[String]]
```

Comment:
Are you planning on coming back to all of the commented out stuff?

Reply:
I suppose @sraaphorst will take those when the time comes